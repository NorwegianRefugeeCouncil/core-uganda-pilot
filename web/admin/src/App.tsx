import React, {Fragment} from 'react';
import './App.scss';
import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap-icons/font/bootstrap-icons.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';
import {AuthProvider} from "oidc-react"
import {NavBar} from "./components/navbar/NavBar";
import {BrowserRouter, Switch} from "react-router-dom";
import {Organizations} from "./components/organizations/Organizations";
import {ProtectedRoute} from "./components/guard/ProtectedRoute";
import {OrganizationPortal} from "./components/organization/OrganizationPortal";
import {OrganizationEditor} from "./components/organization/OrganizationEditor";
import {Clients} from "./components/clients/Clients";

const oidcConfig = {
    authority: 'https://dev-53701279.okta.com',
    clientId: '0oa2gb25jhuFDIkhd5d7',
    redirectUri: 'http://localhost:3001'
};

function App() {
    return (
        <Fragment>
            <AuthProvider
                scope={"openid profile email"}
                autoSignIn={true}
                {...oidcConfig} >
                <div className={"d-flex flex-column vh-100 vw-100 bg-dark"}>
                    <BrowserRouter>
                        <NavBar/>
                        <Switch>
                            <ProtectedRoute path={"/organizations/add"} component={OrganizationEditor}/>
                            <ProtectedRoute path={"/organizations/:organizationId"} component={OrganizationPortal}/>
                            <ProtectedRoute path={"/organizations"} component={Organizations}/>
                            <ProtectedRoute path={"/clients"} component={Clients}/>
                        </Switch>
                    </BrowserRouter>
                </div>
            </AuthProvider>
        </Fragment>
    );
}

export default App;
