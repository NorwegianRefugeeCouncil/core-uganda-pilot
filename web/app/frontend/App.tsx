import React from 'react';
import { IAMClient } from 'core-js-api-client';
import { Provider as PaperProvider } from 'react-native-paper';
import theme from './src/constants/theme';
import Router from './src/components/Router';
import { Subject } from 'rxjs';
import { Platform } from 'react-native';
import Constants from 'expo-constants';

const host = Platform.OS === 'web' ? Constants.manifest?.extra?.server_default_hostname : Constants.manifest?.extra?.server_hostname;
console.log('host: ', host);

const subject = new Subject();

const iamClient = new IAMClient('http', host, { 'X-Authenticated-User-Subject': ['test@user.email'] });

function get() {
	subject.pipe(iamClient.Parties().Get()).subscribe(console.log);
	subject.next('c529d679-3bb6-4a20-8f06-c096f4d9adc1');
}

export default function App() {
	get();
	return (
		<PaperProvider theme={theme}>
			<Router host={host} />
		</PaperProvider>
	);
}


