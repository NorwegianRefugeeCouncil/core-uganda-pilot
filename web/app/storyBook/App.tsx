import React from 'react';
import StorybookUIRoot from "./storybook";

export default function App() {
    return (
        <StorybookUIRoot/>
    );
}
