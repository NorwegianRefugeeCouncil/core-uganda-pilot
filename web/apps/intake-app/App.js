"use strict";
exports.__esModule = true;
var react_1 = require("react");
var react_native_paper_1 = require("react-native-paper");
var theme_1 = require("./src/constants/theme");
var Router_1 = require("./src/components/Router");
var WebBrowser = require("expo-web-browser");
var AuthWrapper_1 = require("./src/components/AuthWrapper");
WebBrowser.maybeCompleteAuthSession();
function App() {
    return (<react_native_paper_1.Provider theme={theme_1["default"]}>
            <AuthWrapper_1.AuthWrapper>
                <Router_1["default"] />
            </AuthWrapper_1.AuthWrapper>
        </react_native_paper_1.Provider>);
}
exports["default"] = App;
