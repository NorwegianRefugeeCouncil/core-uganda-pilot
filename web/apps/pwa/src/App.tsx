import React from 'react';
import 'bootstrap-icons/font/bootstrap-icons.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';
import {BrowserRouter, Route, Switch} from "react-router-dom";
import {AuthWrapper} from "@core/auth-client";
import defaultAuth from "./constants/defaultAuth";
import AuthenticatedApp from "./components/AuthenticatedApp";
import client from "./app/client";

const App: React.FC = () => {
    return (
        <BrowserRouter basename={`/${defaultAuth.redirectUriSuffix}`}>
            <Switch>
                <Route path={""} render={() =>
                    <AuthWrapper
                        clientId={process?.env?.REACT_APP_CLIENT_ID || defaultAuth.clientId}
                        axiosInstance={client.axiosInstance}
                        scopes={defaultAuth.scopes}
                        issuer={process?.env?.REACT_APP_ISSUER || defaultAuth.issuer}
                        redirectUriSuffix={defaultAuth.redirectUriSuffix}
                    >
                        <AuthenticatedApp/>
                    </AuthWrapper>
                }/>
            </Switch>
        </BrowserRouter>
    );
}

export default App;

