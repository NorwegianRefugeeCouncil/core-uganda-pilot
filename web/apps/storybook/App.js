"use strict";
exports.__esModule = true;
var react_1 = require("react");
var storybook_1 = require("./storybook");
function App() {
    return (<storybook_1["default"] />);
}
exports["default"] = App;
