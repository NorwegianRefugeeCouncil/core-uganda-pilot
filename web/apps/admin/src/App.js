"use strict";
exports.__esModule = true;
var react_1 = require("react");
require("./App.scss");
require("bootstrap/dist/css/bootstrap.css");
require("bootstrap-icons/font/bootstrap-icons.css");
require("bootstrap/dist/js/bootstrap.bundle.min");
var react_router_dom_1 = require("react-router-dom");
var AuthenticatedApp_1 = require("./components/AuthenticatedApp");
var auth_client_1 = require("@core/auth-client");
var client_1 = require("../src/client/client");
function App() {
    var _a, _b;
    var client = new client_1["default"]();
    return (<react_router_dom_1.BrowserRouter>
            <react_router_dom_1.Switch>
                <auth_client_1.AuthWrapper clientId={((_a = process === null || process === void 0 ? void 0 : process.env) === null || _a === void 0 ? void 0 : _a.REACT_APP_CLIENT_ID) || ''} scopes={['openid', 'profile', 'offline_access']} issuer={((_b = process === null || process === void 0 ? void 0 : process.env) === null || _b === void 0 ? void 0 : _b.REACT_APP_ISSUER) || ''} redirectUriSuffix={'app'} axiosInstance={client.axiosInstance}>
                    <AuthenticatedApp_1["default"] />
                </auth_client_1.AuthWrapper>
            </react_router_dom_1.Switch>
        </react_router_dom_1.BrowserRouter>);
}
exports["default"] = App;
