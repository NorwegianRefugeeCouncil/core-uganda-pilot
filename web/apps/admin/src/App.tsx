import React from 'react';
import './App.scss';
import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap-icons/font/bootstrap-icons.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';
import {BrowserRouter, Switch} from "react-router-dom";
import AuthenticatedApp from "./components/AuthenticatedApp";
import {AuthWrapper} from "@core/auth-client";
import Client from "../src/client/client";

function App() {
    const client = new Client();
    return (
        <BrowserRouter>
            <Switch>
                <AuthWrapper
                    clientId={process?.env?.REACT_APP_CLIENT_ID || ''}
                    scopes={['openid', 'profile', 'offline_access']}
                    issuer={process?.env?.REACT_APP_ISSUER || ''}
                    redirectUriSuffix={'app'}
                    axiosInstance={client.axiosInstance}
                >
                    <AuthenticatedApp/>
                </AuthWrapper>
            </Switch>
        </BrowserRouter>
    );
}

export default App;
