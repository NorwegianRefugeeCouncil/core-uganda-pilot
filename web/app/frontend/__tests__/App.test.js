import React from 'react';
import Renderer from 'react-test-renderer';
import App from '../App.tsx';

describe('App', () => {
    
    it('<App /> has 1 child', () => {
      const tree = Renderer.create(<App />).toJSON();
      expect(tree.children.length).toBe(1);
    });

    it('renders correctly', () => {
        const tree = Renderer
          .create(<App />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

  });