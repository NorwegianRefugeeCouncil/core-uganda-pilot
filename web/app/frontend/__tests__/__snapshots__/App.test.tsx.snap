// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "rippleRadius": undefined,
        "type": "ThemeAttrAndroid",
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      LOGIN
    </Text>
  </View>
</View>
`;
